packages/adapter-vercel/index.js

```diff
@@ -158,6 +142,22 @@ const plugin = function (defaults = {}) {
 			/** @type {Map<string, string>} */
 			const functions = new Map();
 
+			// every route needs a `group`, so that a page and its __data.json are invalidated
+			// as a pair. automatically assign group ids to routes that don't have one
+			let group_id = 0;
+
+			// ensure automatically assigned group ids are unique
+			for (const route of builder.routes) {
+				if (route.prerender === true) continue;
+
+				if (route.config?.isr?.group !== undefined) {
+					group_id = route.config.isr.group;
```

Comment:
I don't understand the assignment here. Is the idea that this ensures that automatic `group_id` assignment doesn't become an existing group id? I don't think that this is enough then - we would need to store those in a set. We can't be sure that they are strictly incremental.

Reply:
If someone explicitly specifies `group`, we can't use the same value they provided. This is supposed to ensure that `group_id` increments from a starting value that is higher than the largest user-provided value, but I forgot the `Math.max`

Reply:
That said, I _think_ that invalidation only happens to a set of assets with the same `group` if they also have the same query, including `__pathname`. So it's likely that allowing users to set this value will have no effect. Will try and get clarity on that, it may be the case that we should just remove it from the exposed options

Reply:
Alright, conclusion: `group` should not be expose to users, it's an implementation detail for frameworks. Will remove it